c2corg_ui/views/document.py

```diff
@@ -432,12 +441,17 @@ def _redirect_to_full_url(self):
             raise HTTPInternalServerError(
                 "An error occurred while loading the document")
 
-        locale = data['locales'][0]
-        title = ''
-        if self._API_ROUTE == 'routes' and locale['title_prefix']:
-            title += locale['title_prefix'] + ' '
-        title += locale['title']
-        location = self.request.route_url(
-            self._API_ROUTE + '_view', id=data['document_id'],
-            lang=locale['lang'], slug=slugify(title))
-        raise HTTPMovedPermanently(location=location)
+        if 'redirects_to' in data:
+            if lang not in data['available_langs']:
+                lang = self._get_preferred_lang()
```

Comment:
If you call `/waypoints/1234/fr` and there is no locale for `fr` for waypoint `1234`, you will get a `404`.

> What is "the best lang in data['available_langs']"?

https://github.com/c2corg/v6_api/blob/b32316f7ee1a9db28107e20c9678c87a7c235468/c2corg_api/views/__init__.py#L168-L175